src/Sarif.Viewer.VisualStudio/ErrorList/SarifResultTableEntry.cs

```diff
@@ -0,0 +1,218 @@
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Windows;
+    using System.Windows.Documents;
+
+    internal sealed class SarifResultTableEntry : ITableEntry
+    {
+        private readonly Dictionary<string, object> columnKeyToContent = new Dictionary<string, object>(StringComparer.InvariantCulture);
+
+        public static readonly string[] SupportedColumns = {
+            StandardTableKeyNames2.TextInlines,
+            StandardTableKeyNames.DocumentName,
+            StandardTableKeyNames.ErrorCategory,
+            StandardTableKeyNames.Line,
+            StandardTableKeyNames.Column,
+            StandardTableKeyNames.Text,
+            StandardTableKeyNames.FullText,
+            StandardTableKeyNames.ErrorSeverity,
+            StandardTableKeyNames.Priority,
+            StandardTableKeyNames.ErrorSource,
+            StandardTableKeyNames.BuildTool,
+            StandardTableKeyNames.ErrorCode,
+            StandardTableKeyNames.ProjectName,
+            StandardTableKeyNames.HelpLink,
+            StandardTableKeyNames.ErrorCodeToolTip,
+            "suppressionstatus",
+            "suppressionstate",
+            "suppression"
+        };
+
+        public SarifResultTableEntry(SarifErrorListItem error)
+        {
+            this.Identity = error.GetHashCode();
+            this.Error = error;
+
+            // Set the data that's fast to retrieve into the dictionary of values.
+            this.columnKeyToContent[StandardTableKeyNames.DocumentName] = this.Error.FileName;
+            this.columnKeyToContent[StandardTableKeyNames.ErrorCategory] = this.Error.Category;
+
+            // The error list assumes the line number provided will be zero based and adds one before displaying the value.
+            // i.e. if we pass 5, the error list will display 6. 
+            // Subtract one from the line number so the error list displays the correct value.
+            this.columnKeyToContent[StandardTableKeyNames.Line] = this.Error.LineNumber - 1;
+
+            this.columnKeyToContent[StandardTableKeyNames.Column] = this.Error.ColumnNumber;
+            this.columnKeyToContent[StandardTableKeyNames.ErrorSeverity] = GetSeverity(this.Error.Level);
+            this.columnKeyToContent[StandardTableKeyNames.Priority] = GetSeverity(this.Error.Level) == __VSERRORCATEGORY.EC_ERROR
+                    ? vsTaskPriority.vsTaskPriorityHigh
+                    : vsTaskPriority.vsTaskPriorityMedium;
+            this.columnKeyToContent[StandardTableKeyNames.ErrorSource] = ErrorSource.Build;
+            this.columnKeyToContent[StandardTableKeyNames.BuildTool] = this.Error.Tool?.Name;
+
+            if (this.Error.Rule != null)
+            {
+                this.columnKeyToContent[StandardTableKeyNames.ErrorCode] = this.Error.Rule.Id;
+                this.columnKeyToContent[StandardTableKeyNames.ErrorCodeToolTip] = this.Error.Rule.Id + ":" + this.Error.Rule.Name;
```

Comment:
>Name [](start = 125, length = 4)

Be aware that this is optional

Reply:
This isn't fixed. Don't add `"." + Name` unless `Name` is non-null-or-whitespace.

---
In reply to: [474068843](https://github.com/microsoft/sarif-visualstudio-extension/pull/172#discussion_r474068843) [](ancestors = 474068843)